portfolio/src/main/java/com/google/sps/servlets/SpotifyServlet.java

```diff
@@ -0,0 +1,150 @@
+package com.google.sps.servlets;
+
+import com.wrapper.spotify.SpotifyApi;
+import com.wrapper.spotify.SpotifyHttpManager;
+import com.wrapper.spotify.exceptions.SpotifyWebApiException;
+import com.wrapper.spotify.model_objects.credentials.AuthorizationCodeCredentials;
+import com.wrapper.spotify.model_objects.miscellaneous.CurrentlyPlaying;
+import com.wrapper.spotify.model_objects.specification.Track;
+import com.wrapper.spotify.model_objects.IPlaylistItem;
+import com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeRefreshRequest;
+import com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeRequest;
+import com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeUriRequest;
+import com.wrapper.spotify.requests.data.player.GetUsersCurrentlyPlayingTrackRequest;
+import org.apache.hc.core5.http.ParseException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/spotify")
+public class SpotifyServlet extends HttpServlet {
+
+  private static final String clientId = "444dda318d8949cb9282d95a9885a7c6";
+  private static final String clientSecret = "9f81b8a6cf5748ef9d24a7ddb1e8bc4c";
+  private static final String redirectUriString = "https://8080-b0939ef0-d7b7-4da5-abf9-d0bf54e01dcf.us-east1.cloudshell.dev/spotify";
```

Comment:
Does this rely on you leaving your cloudshell server running? That might be annoying

Reply:
For now I have the redirect URI as my CloudShell devserver so that I can test if it's working, but once the PR gets approved, it would be changed to the deployed website instead.